src/controllers/index.ts

```diff
@@ -0,0 +1,63 @@
+import { Response, Request } from 'express';
+import { logger } from '../services';
+import { generateRandomString, HTTP_STATUS } from '../utils';
+import URL_DB from '../model/url';
+
+const shortenUrl = async (req: Request, res: Response) => {
+  try {
+    const { url, customName } = req.body;
+
+    const randomName = customName ? customName.replace(/\s/g, '-') : generateRandomString();
+
+    const existingUrl = await URL_DB.findOne({ originalUrl: url });
+
+    if (existingUrl) {
+      return res.status(HTTP_STATUS.CONFLICT.code).send({
+        message: 'URL already shortened',
+        data: { url, customName: existingUrl.customName }
+      });
```

Comment:
Return an appropiate error message
```suggestion
      return res.status(HTTP_STATUS.CONFLICT.code).send({
        message: 'Conflict'
      });
```